oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/core/oidc/OidcProviderConfiguration.java

```diff
@@ -118,6 +119,15 @@ public Builder idTokenSigningAlgorithms(Consumer<List<String>> signingAlgorithms
 			return this;
 		}
 
+		/**
+		 * Add this registration endpoint in the resulting {@link OidcProviderConfiguration} OPTIONAL
+		 * @param registrationEndpoint the supported registration client url
+		 * @return the {@link Builder} for further configuration
+		 */
+		public Builder registrationEndpoint(String registrationEndpoint) {
```

Comment:
This should get pushed up to `AbstractOAuth2AuthorizationServerMetadata.AbstractBuilder` as `registration_endpoint` is also available in [OAuth 2.0 Authorization Server Metadata](https://datatracker.ietf.org/doc/html/rfc8414#section-2).

Also, let's rename the method to `clientRegistrationEndpoint()`

Reply:
Hi @jgrandja,
Currently, the `OAuth2AuthorizationServerMetadataEndpointFilter` does not expose this. Is it ok if I work on it? And if so can use the same PR or should I create a new one.

Reply:
gh-56 is on hold as we don't need the feature as of yet.